### PR TITLE
use builtin so as to not break aliases in shell integration script

### LINKED ISSUE #145580: shell integration does not work when rc define aliases for certain things
>
- Breaks when user rc files define aliases for things like `history` or global aliases for things like `-n`.
_Originally posted by @romkatv in https://github.com/microsoft/vscode/issues/142004#issuecomment-1073263418_

### PR DESCRIPTION
**Pull request by meganrogge**
This PR fixes #145580

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/media/shellIntegration.zsh [hunk 1]
```diff
@@ -25,50 +25,50 @@ fi
...
 }
 
 __vsc_right_prompt_end() {
-	printf "\033]633;I\007"
+	builtin printf "\033]633;I\007"
 }
 
 __vsc_command_complete() {
-	local __vsc_history_id=$(history | tail -n1 | awk '{print $1;}')
+	builtin local __vsc_history_id=$(builtin history | tail -n1 | awk '{print $1;}')
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/media/shellIntegration.zsh:67]: Same, is builtin needed here?

File: src/vs/workbench/contrib/terminal/browser/media/shellIntegration.zsh [hunk 2]
```diff
@@ -25,50 +25,50 @@ fi
 # Shell integration was disabled by the shell, exit without warning assuming either the shell has
 # explicitly disabled shell integration as it's incompatible or it implements the protocol.
 if [ -z "$VSCODE_SHELL_INTEGRATION" ]; then
-	return
+	builtin return
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/media/shellIntegration.zsh:28]: I believe `return` would be a reserved keyword not a builtin so this should be safe (and may break doing this).

**Review by Tyriar — CHANGES_REQUESTED** [review]
Haven't tested but looks like what I'd think. I believe bash also has this feature though, we'll want to cover that too.

File: src/vs/workbench/contrib/terminal/browser/media/shellIntegration.zsh (hunk 1 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/media/shellIntegration.zsh:67]: kitty does that, we should too

File: src/vs/workbench/contrib/terminal/browser/media/shellIntegration.zsh (hunk 2 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/media/shellIntegration.zsh:28]: kitty does this and it seems to work

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/media/shellIntegration.zsh:28]: https://github.com/kovidgoyal/kitty/blob/master/shell-integration/zsh/kitty-integration#L26

File: src/vs/workbench/contrib/terminal/browser/media/shellIntegration.zsh (hunk 1 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/media/shellIntegration.zsh:67]: https://github.com/kovidgoyal/kitty/blob/master/shell-integration/zsh/kitty-integration#L66

**Review by Tyriar — APPROVED** [review]
Looks good if tested, also does `echo` and `printf` need this treatment in bash too?

**Review by raghavthind2005 — APPROVED** [review]

**Pull request merged**